lib/browser/api/web-contents.ts

```diff
@@ -67,8 +75,8 @@ const PDFPageSizes = {
 // Default printing setting
 const defaultPrintingSetting = {
```

Comment:
Instead of all the `as` casts below we should type this as `Electron.WhateverThePrintingSettingsInterfaceIs`

Reply:
I don't think there exists any type in electron.d.ts that matches this.

```
$ rg '\bpageRange\b'
lib/browser/api/web-contents.ts
78:  pageRange: [] as {from: number, to: number}[],
279:    printSettings.pageRange = [{
$
```

Reply:
@nornagon That's because according to the docs it's `pageRanges`

Reply:
This is also true of several other members of this struct:

```
$ rg '\bshouldPrintBackgrounds\b'
lib/browser/api/web-contents.ts
84:  shouldPrintBackgrounds: false,
258:    printSettings.shouldPrintBackgrounds = options.printBackground;
$ rg '\bheaderFooterEnabled\b'
lib/browser/api/web-contents.ts
81:  headerFooterEnabled: false,
287:    printSettings.headerFooterEnabled = true;
$ rg '\bgenerateDraftData\b'
lib/browser/api/web-contents.ts
96:  generateDraftData: true,
```

I don't think the struct you're thinking of is the same as this struct.

Reply:
I believe the type of this struct is related to [these](https://source.chromium.org/chromium/chromium/src/+/master:printing/print_job_constants.h;l=15-77;drc=e454f2607b49d8eee3f5d40d09c3d4b9c0dd2572)